core/imt/src/TThreadExecutor.cxx

```diff
@@ -153,19 +153,25 @@ namespace ROOT {
 
    void TThreadExecutor::ParallelFor(unsigned int start, unsigned int end, unsigned step, const std::function<void(unsigned int i)> &f)
    {
-      tbb::this_task_arena::isolate([&]{
-         tbb::parallel_for(start, end, step, f);
+      fSched->Arena()->execute([&]{
+         tbb::this_task_arena::isolate([&]{
+            tbb::parallel_for(start, end, step, f);
+         });
```

Comment:
just to make sure I understand correctly: the wrapping within `fSched->Arena()->execute` is to make sure that `tbb::this_task_arena::isolate` picks up ROOT's task arena and not another?

Reply:
Yes.